spec/mongoid/traversable_spec.rb

```diff
@@ -251,4 +251,328 @@
+
+      it "sets the child discriminator key to _type: Guitar" do
+        expect(Guitar.discriminator_key).to eq("_type")
+      end
+    end
+
+    context "when the discriminator key is changed at the global level" do
+      before do
+        Mongoid.discriminator_key = "hello"
+      end
+
+      after do
+        Mongoid.discriminator_key = "_type"
+      end
+
+      it "sets the correct value globally" do
+        expect(Mongoid.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the parent" do
+        expect(Instrument.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the child: Piano" do
+        expect(Piano.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the child: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello")
+      end
+    end
+
+    context "when the discriminator key is changed in the parent" do
+      before do
+        Instrument.discriminator_key = "hello2"
+      end
+
+      after do
+        Instrument.discriminator_key = nil
+      end
+
+      it "doesn't change the global setting" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "changes in the parent" do
+        expect(Instrument.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Piano" do
+        expect(Piano.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello2")
+      end
+
+      context 'when discriminator key is set to nil in parent' do
+        before do
+          Instrument.discriminator_key = nil
+        end
+
+        it "doesn't change the global setting" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'uses global setting' do
+          expect(Instrument.discriminator_key).to eq("_type")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("_type")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("_type")
+        end
+      end
+
+      context "when resetting the discriminator key after nil" do
+        before do
+          Instrument.discriminator_key = nil
+          Instrument.discriminator_key = "hello4"
+        end
+
+        it "doesn't change the global setting" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'has the correct value' do
+          expect(Instrument.discriminator_key).to eq("hello4")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("hello4")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("hello4")
+        end
+      end
+    end
+
+    context "when the discriminator key is changed in the child" do
+      let(:guitar) do 
```

Comment:
Could you give this helper a slightly more descriptive name like `set_discriminator_key`?